From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Valtteri Koskivuori <vkoskiv@gmail.com>
Date: Sat, 20 Apr 2024 01:46:08 +0300
Subject: [PATCH] 6437: M68000: Implement BKPT opcode

Added a new __m68k_bkpt() pcode to indicate it is a CPU intrinsic
---
 Ghidra/Processors/68000/data/languages/68000.sinc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Ghidra/Processors/68000/data/languages/68000.sinc b/Ghidra/Processors/68000/data/languages/68000.sinc
index 407ff99450..505e50a5d2 100644
--- a/Ghidra/Processors/68000/data/languages/68000.sinc
+++ b/Ghidra/Processors/68000/data/languages/68000.sinc
@@ -362,6 +362,7 @@ attach values quick [ 8 1 2 3 4 5 6 7 ];
 define pcodeop kfactor;
 define pcodeop ftrap;
 define pcodeop __m68k_trap;
+define pcodeop __m68k_bkpt;
 define pcodeop __m68k_trapv;
 define pcodeop reset;
 define pcodeop saveFPUStateFrame;
@@ -1152,7 +1153,7 @@ with : extGUARD=1 {
 }
 @endif # CPU32
 
-:bkpt "#"op02			is opbig=0x48 & op67=1 & op5=0 & op34=1 & op02			unimpl
+:bkpt "#"^op02			is opbig=0x48 & op67=1 & op5=0 & op34=1 & op02	{ breakpoint:1 = op02; __m68k_bkpt(breakpoint); }
 
 :bra.b addr8			is opbig=0x60 & addr8						{ goto addr8; }
 :bra.w addr16			is opbig=0x60 & d8base=0; addr16				{ goto addr16; }
-- 
2.44.0

